### ISSUE TITLE
DType conversion

### ISSUE THREAD

**Issue by lzehl** [G-Node/python-odml#96]
I realized that it is not possible anymore to convert from a metadata entry 1(odml.DType.int or .float) to True (odml.DType.boolean). Is there a reason why this is not possible anymore? I think it would be better if this would work. Your thoughts?

**Comment by jgrewe** [comment]
Hi Lyuba, are you sure it worked before with py3? The conversion code itself should not have been affected but I need to check.

**Comment by digaru19** [comment]
@lzehl  Can you please elaborate just a bit more regarding the issue or the required functionality ?
As per the updated methods (with the removal of `odml.Value` class) , all the `value`s (homogeneous) of a `property` are being kept in a list, and the addition of any other `DType` to the list raises an error. 

(@jgrewe Correct me if I'm wrong )

**Comment by jgrewe** [comment]
You should be able to create a property with a float 1.0 value and later set the dtype to bool

```
p = odml.Property("name", 1.0)
p.dtype = odml.dtype.boolean
```
If I understand it correctly, this fails

**Comment by lzehl** [comment]
@jgrewe and @digaru19: Yes it is exactly what I mean.
`p = odml.Property('n', 1.0)`
Or
`p = odml.Property('n', 1)`
Should be convertable to
`p.dtype = odml.DType.boolean`

@jgrewe to be honest I don't know if this ever worked with py3.

**Comment by digaru19** [comment]
@jgrewe Cool .. I will work on it .. :+1: 

@jgrewe @lzehl  Some clarifications :- 
- If it is a list of values, should the whole list be converted to corresponding `bool` type ? 
- Should values other than `int` and `float` also be changed to `bool` type ?

**Comment by jgrewe** [comment]
@digaru19 yes, if a new dtype is set the whole list must be checked. It should be always homogeneous, anyway. The dtype setter actually does this but in the case of the boolean dtype runs into an exception and thus throws a ValueError. I think it is fair to use the bool cast that treats everything that is not zeros as TRUE. 
Please add a test case to the test_property.py, this would give us some more coverage :)

**Comment by lzehl** [comment]
@digaru19 I agree with @jgrewe. Thanks for taking care of this :)

**Comment by jgrewe** [comment]
solved via #99
